#### Relatório de labels que estão erradas 

In [3]:
import pandas as pd

In [6]:
label  = pd.read_csv('../Datahub/label_dataframe.csv').drop(columns = ['Unnamed: 0'])

In [1]:
##### label['weight'] = (label['rmsd_weight']*0.5 + label['similarity_weight']*0.5)/10

In [7]:
label.head()

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  weight  \
0  4eor         1   16.666667  4.34          0.0                0.0    0.00   
1  4eor         2   72.916667  1.16         10.0                7.0    0.85   
2  4eor         3    8.333333  4.94          0.0                0.0    0.00   
3  4eor         4    7.291667  4.92          0.0                0.0    0.00   
4  4eor         5   18.750000  6.54          0.0                0.0    0.00   

   weight_similarity  weight_rmsd  label  
0               0.00         0.00      0  
1               0.79         0.91      1  
2               0.00         0.00      0  
3               0.00         0.00      0  
4               0.00         0.00      0

In [8]:
label[label['pdb'] =='1c5z']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
1730  1c5z         1   60.759494  0.37         10.0                0.0   
1731  1c5z         2   48.101266  0.29         10.0                0.0   
1732  1c5z         3   34.177215  0.34         10.0                0.0   
1733  1c5z         4    8.860759  3.59          0.0                0.0   
1734  1c5z         5    1.265823  4.11          0.0                0.0   
1735  1c5z         6    7.594937  3.95          0.0                0.0   
1736  1c5z         7    3.797468  3.39          0.0                0.0   
1737  1c5z         8   18.987342  3.39          0.0                0.0   
1738  1c5z         9   12.658228  3.18          0.0                0.0   
1739  1c5z        10    0.000000  3.17          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
1730     0.5                0.3          0.7      0  
1731     0.5                0.3          0.7      0  
1732     0.5                0.3          0.7      0  
1733     0.0                0.0          0.0      0  
1734     0.0                0.0          0.0      0  
1735     0.0                0.0          0.0      0  
1736     0.0                0.0          0.0      0  
1737     0.0                0.0          0.0      0  
1738     0.0                0.0          0.0      0  
1739     0.0                0.0          0.0      0

* Similaridade baixa 60.75, 48.10, 34.17 ??? será os hidrogênios que estão faltando ? 

In [16]:
label[label['pdb'] =='1bcu']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
2270  1bcu         1   92.727273  0.48         10.0               10.0   
2271  1bcu         2    0.000000  2.66          7.0                0.0   
2272  1bcu         3    5.454545  0.40         10.0                0.0   
2273  1bcu         4   34.545455  2.67          7.0                0.0   
2274  1bcu         5    1.818182  1.30         10.0                0.0   
2275  1bcu         6    0.000000  3.08          0.0                0.0   
2276  1bcu         7   29.090909  2.80          7.0                0.0   
2277  1bcu         8   54.545455  2.01          7.0                0.0   
2278  1bcu         9    1.818182  3.73          0.0                0.0   
2279  1bcu        10    0.000000  3.45          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
2270    1.00               1.00         1.00      1  
2271    0.35               0.21         0.49      0  
2272    0.50               0.30         0.70      0  
2273    0.35               0.21         0.49      0  
2274    0.50               0.30         0.70      0  
2275    0.00               0.00         0.00      0  
2276    0.35               0.21         0.49      0  
2277    0.35               0.21         0.49      0  
2278    0.00               0.00         0.00      0  
2279    0.00               0.00         0.00      0

* Similarity score está errado para a pose 3 !?

In [17]:
label[label['pdb'] =='1bzc']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
2130  1bzc         1   66.942149  0.62         10.0                0.0   
2131  1bzc         2   88.429752  0.90         10.0               10.0   
2132  1bzc         3   63.636364  1.30         10.0                0.0   
2133  1bzc         4   62.809917  2.00          7.0                0.0   
2134  1bzc         5   55.371901  2.98          7.0                0.0   
2135  1bzc         6   55.371901  2.69          7.0                0.0   
2136  1bzc         7   16.528926  3.16          0.0                0.0   
2137  1bzc         8   76.859504  3.87          0.0                7.0   
2138  1bzc         9   33.057851  2.23          7.0                0.0   
2139  1bzc        10    0.000000  2.76          7.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
2130    0.50               0.30         0.70      0  
2131    1.00               1.00         1.00      1  
2132    0.50               0.30         0.70      0  
2133    0.35               0.21         0.49      0  
2134    0.35               0.21         0.49      0  
2135    0.35               0.21         0.49      0  
2136    0.00               0.00         0.00      0  
2137    0.35               0.49         0.21      0  
2138    0.35               0.21         0.49      0  
2139    0.35               0.21         0.49      0

* A similaridade de 1 era para estar maior que a 2 e não menor 

In [19]:
label[label['pdb'] =='1e66']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  weight  \
660  1e66         1   79.381443  0.28         10.0                7.0    0.85   
661  1e66         2    0.000000   NaN          NaN                0.0     NaN   
662  1e66         3    0.000000   NaN          NaN                0.0     NaN   
663  1e66         4    0.000000   NaN          NaN                0.0     NaN   
664  1e66         5    0.000000   NaN          NaN                0.0     NaN   
665  1e66         6    0.000000   NaN          NaN                0.0     NaN   
666  1e66         7    0.000000   NaN          NaN                0.0     NaN   
667  1e66         8    0.000000   NaN          NaN                0.0     NaN   
668  1e66         9    0.000000   NaN          NaN                0.0     NaN   
669  1e66        10    0.000000   NaN          NaN                0.0     NaN   

     weight_similarity  weight_rmsd  label  
660               0.79         0.91      1  
661                NaN          NaN      0  
662                NaN          NaN      0  
663                NaN          NaN      0  
664                NaN          NaN      0  
665                NaN          NaN      0  
666                NaN          NaN      0  
667                NaN          NaN      0  
668                NaN          NaN      0  
669                NaN          NaN      0

<font color = 'green'> Correto! 

In [21]:
label[label['pdb'] =='1eby']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
1780  1eby         1   82.278481  0.37         10.0               10.0   
1781  1eby         2    8.860759  0.76         10.0                0.0   
1782  1eby         3    5.696203  1.88         10.0                0.0   
1783  1eby         4    5.696203  1.92         10.0                0.0   
1784  1eby         5   40.506329  1.97         10.0                0.0   
1785  1eby         6   38.607595  1.96         10.0                0.0   
1786  1eby         7   19.620253  2.70          7.0                0.0   
1787  1eby         8   51.265823  4.19          0.0                0.0   
1788  1eby         9    6.329114  4.21          0.0                0.0   
1789  1eby        10    0.000000  4.41          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
1780    1.00               1.00         1.00      1  
1781    0.50               0.30         0.70      0  
1782    0.50               0.30         0.70      0  
1783    0.50               0.30         0.70      0  
1784    0.50               0.30         0.70      0  
1785    0.50               0.30         0.70      0  
1786    0.35               0.21         0.49      0  
1787    0.00               0.00         0.00      0  
1788    0.00               0.00         0.00      0  
1789    0.00               0.00         0.00      0

<font color = 'red'> Os hidrogênios estão sendo considerados? Porque a pose 2 é tão pior que a 1?  

In [23]:
label[label['pdb'] =='1g2k']

pdb  poserank  similarity   RMSD  rmsd_weight  similarity_weight  \
220  1g2k         1   76.119403   0.55         10.0                7.0   
221  1g2k         2   67.910448   1.60         10.0                0.0   
222  1g2k         3   22.388060  10.29          0.0                0.0   
223  1g2k         4   17.164179   8.12          0.0                0.0   
224  1g2k         5    8.208955  10.66          0.0                0.0   
225  1g2k         6   23.880597   6.68          0.0                0.0   
226  1g2k         7    1.492537   5.25          0.0                0.0   
227  1g2k         8   18.656716   7.87          0.0                0.0   
228  1g2k         9    2.238806   5.37          0.0                0.0   
229  1g2k        10    0.000000   5.44          0.0                0.0   

     weight  weight_similarity  weight_rmsd  label  
220    0.85               0.79         0.91      1  
221    0.50               0.30         0.70      0  
222    0.00               0.00         0.00      0  
223    0.00               0.00         0.00      0  
224    0.00               0.00         0.00      0  
225    0.00               0.00         0.00      0  
226    0.00               0.00         0.00      0  
227    0.00               0.00         0.00      0  
228    0.00               0.00         0.00      0  
229    0.00               0.00         0.00      0

<font color = 'yellow'> Pensar mais nesse!

In [25]:
label[label['pdb'] =='1gpk']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
1350  1gpk         1   83.615819  0.25         10.0               10.0   
1351  1gpk         2   17.514124  3.62          0.0                0.0   
1352  1gpk         3   15.819209  4.75          0.0                0.0   
1353  1gpk         4   23.728814  3.52          0.0                0.0   
1354  1gpk         5   12.429379  4.88          0.0                0.0   
1355  1gpk         6   49.717514  1.42         10.0                0.0   
1356  1gpk         7   17.514124  4.67          0.0                0.0   
1357  1gpk         8   15.254237  4.86          0.0                0.0   
1358  1gpk         9   16.384181  3.93          0.0                0.0   
1359  1gpk        10    0.000000   NaN          NaN                0.0   

      weight  weight_similarity  weight_rmsd  label  
1350     1.0                1.0          1.0      1  
1351     0.0                0.0          0.0      0  
1352     0.0                0.0          0.0      0  
1353     0.0                0.0          0.0      0  
1354     0.0                0.0          0.0      0  
1355     0.5                0.3          0.7      0  
1356     0.0                0.0          0.0      0  
1357     0.0                0.0          0.0      0  
1358     0.0                0.0          0.0      0  
1359     NaN                NaN          NaN      0

<font color = 'green'> __PERFEITO!__

In [27]:
label[label['pdb'] =='1gpn']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
1820  1gpn         1   11.392405  4.43          0.0                0.0   
1821  1gpn         2   83.544304  0.40         10.0               10.0   
1822  1gpn         3   16.455696  3.86          0.0                0.0   
1823  1gpn         4   12.658228  4.60          0.0                0.0   
1824  1gpn         5    3.797468  4.45          0.0                0.0   
1825  1gpn         6   49.367089  1.40         10.0                0.0   
1826  1gpn         7    3.797468  4.27          0.0                0.0   
1827  1gpn         8    8.860759  4.44          0.0                0.0   
1828  1gpn         9   20.253165  3.95          0.0                0.0   
1829  1gpn        10    0.000000   NaN          NaN                0.0   

      weight  weight_similarity  weight_rmsd  label  
1820     0.0                0.0          0.0      0  
1821     1.0                1.0          1.0      1  
1822     0.0                0.0          0.0      0  
1823     0.0                0.0          0.0      0  
1824     0.0                0.0          0.0      0  
1825     0.5                0.3          0.7      0  
1826     0.0                0.0          0.0      0  
1827     0.0                0.0          0.0      0  
1828     0.0                0.0          0.0      0  
1829     NaN                NaN          NaN      0

<font color = 'green'> __BOM!__

In [30]:
label[label['pdb'] =='1h22']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
1050  1h22         1    4.255319  6.42          0.0                0.0   
1051  1h22         2   46.808511  6.38          0.0                0.0   
1052  1h22         3    2.127660  3.46          0.0                0.0   
1053  1h22         4   27.234043   NaN          NaN                0.0   
1054  1h22         5   61.276596  1.65         10.0                0.0   
1055  1h22         6    0.851064  2.76          7.0                0.0   
1056  1h22         7    2.127660  2.88          7.0                0.0   
1057  1h22         8   50.638298  3.03          0.0                0.0   
1058  1h22         9    0.851064   NaN          NaN                0.0   
1059  1h22        10    0.000000  3.96          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
1050    0.00               0.00         0.00      0  
1051    0.00               0.00         0.00      0  
1052    0.00               0.00         0.00      0  
1053     NaN                NaN          NaN      0  
1054    0.50               0.30         0.70      0  
1055    0.35               0.21         0.49      0  
1056    0.35               0.21         0.49      0  
1057    0.00               0.00         0.00      0  
1058     NaN                NaN          NaN      0  
1059    0.00               0.00         0.00      0

<font color = 'green'> __BOM!__

In [31]:
list(label.sort_values('pdb')['pdb'].unique())[10:20]

['1h23',
 '1k1i',
 '1lpg',
 '1mq6',
 '1nc1',
 '1nc3',
 '1nvq',
 '1o0h',
 '1o3f',
 '1o5b']

In [32]:
label[label['pdb'] =='1h23']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
2590  1h23         1    4.526749  6.49          0.0                0.0   
2591  1h23         2   39.094650  6.43          0.0                0.0   
2592  1h23         3   13.580247  4.12          0.0                0.0   
2593  1h23         4    1.646091  2.84          7.0                0.0   
2594  1h23         5   25.514403  2.95          7.0                0.0   
2595  1h23         6   67.489712  1.30         10.0                0.0   
2596  1h23         7    1.234568  5.61          0.0                0.0   
2597  1h23         8    9.053498  4.57          0.0                0.0   
2598  1h23         9   10.288066  4.33          0.0                0.0   
2599  1h23        10    0.000000  4.25          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
2590    0.00               0.00         0.00      0  
2591    0.00               0.00         0.00      0  
2592    0.00               0.00         0.00      0  
2593    0.35               0.21         0.49      0  
2594    0.35               0.21         0.49      0  
2595    0.50               0.30         0.70      0  
2596    0.00               0.00         0.00      0  
2597    0.00               0.00         0.00      0  
2598    0.00               0.00         0.00      0  
2599    0.00               0.00         0.00      0

<font color ='yellow'> __Será que a pose 6 não deveria estar como positiva ?__

In [33]:
label[label['pdb'] =='1k1i']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  weight  \
840  1k1i         1   75.000000  4.73          0.0                7.0    0.35   
841  1k1i         2   79.166667  2.00          7.0                7.0    0.70   
842  1k1i         3   50.000000  4.95          0.0                0.0    0.00   
843  1k1i         4   55.833333  4.51          0.0                0.0    0.00   
844  1k1i         5   34.166667  3.30          0.0                0.0    0.00   
845  1k1i         6   50.833333  3.56          0.0                0.0    0.00   
846  1k1i         7   51.666667  5.67          0.0                0.0    0.00   
847  1k1i         8    1.666667  6.36          0.0                0.0    0.00   
848  1k1i         9    0.833333  6.38          0.0                0.0    0.00   
849  1k1i        10    0.000000  4.40          0.0                0.0    0.00   

     weight_similarity  weight_rmsd  label  
840               0.49         0.21      0  
841               0.70         0.70      1  
842               0.00         0.00      0  
843               0.00         0.00      0  
844               0.00         0.00      0  
845               0.00         0.00      0  
846               0.00         0.00      0  
847               0.00         0.00      0  
848               0.00         0.00      0  
849               0.00         0.00      0

<font color = 'green'> O caso da pose 1 é importante, pois a similaridade deu alta, pois a parte que importa está apontada para londe da proteína. 

In [35]:
label[label['pdb'] =='1lpg']

pdb  poserank  similarity   RMSD  rmsd_weight  similarity_weight  \
1830  1lpg         1   20.689655  10.17          0.0                0.0   
1831  1lpg         2    0.000000   9.07          0.0                0.0   
1832  1lpg         3   36.781609   1.03         10.0                0.0   
1833  1lpg         4   19.540230  10.23          0.0                0.0   
1834  1lpg         5   11.494253  10.42          0.0                0.0   
1835  1lpg         6    0.000000   8.64          0.0                0.0   
1836  1lpg         7    0.574713   8.21          0.0                0.0   
1837  1lpg         8   24.137931  10.03          0.0                0.0   
1838  1lpg         9   54.597701   2.51          7.0                0.0   
1839  1lpg        10    0.000000  10.55          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
1830    0.00               0.00         0.00      0  
1831    0.00               0.00         0.00      0  
1832    0.50               0.30         0.70      0  
1833    0.00               0.00         0.00      0  
1834    0.00               0.00         0.00      0  
1835    0.00               0.00         0.00      0  
1836    0.00               0.00         0.00      0  
1837    0.00               0.00         0.00      0  
1838    0.35               0.21         0.49      0  
1839    0.00               0.00         0.00      0

<font color = 'red'> Pose 3 não entrou! Deveria ? E sobre a pose 9? 

In [39]:
label[label['pdb'] =='1mq6']

pdb  poserank  similarity   RMSD  rmsd_weight  similarity_weight  \
1860  1mq6         1   57.723577   1.40         10.0                0.0   
1861  1mq6         2   53.658537   1.42         10.0                0.0   
1862  1mq6         3   57.723577   1.64         10.0                0.0   
1863  1mq6         4   47.154472   2.64          7.0                0.0   
1864  1mq6         5   61.788618   2.01          7.0                0.0   
1865  1mq6         6    5.691057   7.94          0.0                0.0   
1866  1mq6         7    0.813008   8.57          0.0                0.0   
1867  1mq6         8    4.065041   9.37          0.0                0.0   
1868  1mq6         9    8.130081   6.95          0.0                0.0   
1869  1mq6        10    0.000000  10.40          0.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
1860    0.50               0.30         0.70      0  
1861    0.50               0.30         0.70      0  
1862    0.50               0.30         0.70      0  
1863    0.35               0.21         0.49      0  
1864    0.35               0.21         0.49      0  
1865    0.00               0.00         0.00      0  
1866    0.00               0.00         0.00      0  
1867    0.00               0.00         0.00      0  
1868    0.00               0.00         0.00      0  
1869    0.00               0.00         0.00      0

In [40]:
label[label['pdb'] =='1bzc']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  \
2130  1bzc         1   66.942149  0.62         10.0                0.0   
2131  1bzc         2   88.429752  0.90         10.0               10.0   
2132  1bzc         3   63.636364  1.30         10.0                0.0   
2133  1bzc         4   62.809917  2.00          7.0                0.0   
2134  1bzc         5   55.371901  2.98          7.0                0.0   
2135  1bzc         6   55.371901  2.69          7.0                0.0   
2136  1bzc         7   16.528926  3.16          0.0                0.0   
2137  1bzc         8   76.859504  3.87          0.0                7.0   
2138  1bzc         9   33.057851  2.23          7.0                0.0   
2139  1bzc        10    0.000000  2.76          7.0                0.0   

      weight  weight_similarity  weight_rmsd  label  
2130    0.50               0.30         0.70      0  
2131    1.00               1.00         1.00      1  
2132    0.50               0.30         0.70      0  
2133    0.35               0.21         0.49      0  
2134    0.35               0.21         0.49      0  
2135    0.35               0.21         0.49      0  
2136    0.00               0.00         0.00      0  
2137    0.35               0.49         0.21      0  
2138    0.35               0.21         0.49      0  
2139    0.35               0.21         0.49      0

<font color = 'red'> "1c5z" PROBLEMA NESSE"

In [47]:
label[label['pdb'] =='1yc1']

pdb  poserank  similarity  RMSD  rmsd_weight  similarity_weight  weight  \
820  1yc1         1   66.666667  0.79         10.0                0.0     0.5   
821  1yc1         2   51.612903  1.26         10.0                0.0     0.5   
822  1yc1         3    5.376344  6.09          0.0                0.0     0.0   
823  1yc1         4   10.752688  5.29          0.0                0.0     0.0   
824  1yc1         5    8.602151  5.87          0.0                0.0     0.0   
825  1yc1         6    0.000000  4.72          0.0                0.0     0.0   
826  1yc1         7    3.225806  5.66          0.0                0.0     0.0   
827  1yc1         8    6.451613  5.43          0.0                0.0     0.0   
828  1yc1         9    0.000000  5.41          0.0                0.0     0.0   
829  1yc1        10    0.000000   NaN          NaN                0.0     NaN   

     weight_similarity  weight_rmsd  label  
820                0.3          0.7      0  
821                0.3          0.7      0  
822                0.0          0.0      0  
823                0.0          0.0      0  
824                0.0          0.0      0  
825                0.0          0.0      0  
826                0.0          0.0      0  
827                0.0          0.0      0  
828                0.0          0.0      0  
829                NaN          NaN      0